tensorflow/python/kernel_tests/matmul_op_test.py

```diff
@@ -63,29 +69,42 @@ def _GetTransposedMatrices(x, x_name, kwargs):
   else:
     return x
 
-
 class MatMulTest(test_lib.TestCase):
-  pass  # Filled in below
-
+  pass
 
 def _GetMatMulTest(a_np_, b_np_, use_static_shape_, **kwargs_):
 
   @test_util.run_without_tensor_float_32("Tests matmul")
   def Test(self):
-    np_val = np.matrix(a_np_) * np.matrix(b_np_)
+    if a_np_.dtype is dtypes.bfloat16:
+      # converting it back to float32 to avoid precision error
+      a_np_fp32 = np.array(math_ops.cast(a_np_, dtypes.float32))
+      b_np_fp32 = np.array(math_ops.cast(b_np_, dtypes.float32))
```

Comment:
Once we use `.as_numpy_dtype`, we can just call `as_type` here.
```suggestion
      a_np_fp32 = a_np_.as_type(np.float32)
      b_np_fp32 = b_np_.as_type(np.float32)
```
Same for Gradient.